aggregator.py

```diff
@@ -497,6 +502,47 @@ def parse_event_packet(self, packet):
         except (IndexError, ValueError):
             raise Exception(u'Unparseable event packet: %s' % packet)
 
+    def parse_sc_packet(self, packet):
+        try:
+            _, data_and_metadata = packet.split('|', 1)
+            # Service check syntax:
+            # _sc|check_name|status|meta
+            if data_and_metadata.count('|') == 1:
+                # Case with no metadata
+                check_name, status = data_and_metadata.split('|')
+                metadata = ''
+            else:
+                check_name, status, metadata = data_and_metadata.split('|', 2)
+
+            service_check = {
+                'check_name': check_name,
+                'status': int(status)
+            }
+
+            message_delimiter = '|m:' if '|m:' in metadata else 'm:'
+            if message_delimiter in metadata:
+                meta, message = metadata.rsplit(message_delimiter, 1)
+                service_check['message'] = self._unescape_sc_content(message)
+            else:
+                meta = metadata
+
+            if not meta:
+                return service_check
+
+            meta = unicode(meta)
+            for m in meta.split('|'):
+                if m[0] == u'd':
+                    service_check['timestamp'] = float(m[2:])
+                elif m[0] == u'h':
+                    service_check['hostname'] = m[2:]
+                elif m[0] == u'#':
+                    service_check['tags'] = sorted(m[1:].split(u','))
+
+            return service_check
+
+        except IndexError, ValueError:
```

Comment:
Watchout this won't do what you expect.

By doing this you are assigning a caught IndexError exception to ValueError.

```python
Python 2.7.6 (default, Mar 22 2014, 22:59:56)
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> try:
...     l = []
...     print l[1]
... except IndexError, ValueError:
...     print ValueError
...
list index out of range
```
You want to do:
```python
except (IndexError, ValueError)
``` 

which would do The Right Thing (tm):
```python
Python 2.7.6 (default, Mar 22 2014, 22:59:56)
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> try:
...     l = []
...     print l[1]
... except (IndexError, ValueError):
...     print ValueError
...
<type 'exceptions.ValueError'>
```